misc/mmt/mmt.1
.ds dP /usr/pub
.TH MMT 1
.SH NAME
.B mmt
\- typeset document with
.SM MM
macros
.SH SYNOPSIS
\*(mBmmt\f1
.OP "" options []
.OP "" files []
.SH DESCRIPTION
The
.B mmt
command formats input using the
.BR troff (1)
text formatter
and the
.SM MM
macro package.
There are options to specify
preprocessing by
.BR eqn (1),
.BR tbl (1),
.BR pic (1),
and/or
.BR grap (1).
The proper pipelines and the
required arguments and flags for
.B troff
and for the macro package
are generated, depending on the options selected.
.PP
.I Options\^
specific to 
.B mmt
are given below.
Any other arguments or flags (e.g.,
.OP -o list )
are passed to
.BR troff .
You can put options in any order,
but they must appear before any
.IR files .
If you give no arguments or 
.IR files ,
.B mmt
will print a list of its options.
.PP
.PD 0
.TP 9
.OP \-e
invokes
.BR eqn ;
also causes
.B eqn
to read the
.MW \*(dP/eqnchar
file (see
.BR eqnchar (5)).
.TP
.OP \-t
invokes
.BR tbl .
.TP
.OP \-p
invokes
.BR pic .
.TP
.OP \-g
invokes
.BR grap ,
which in turn calls
.BR pic .
.TP
.OP \-T type
creates output for
.BR troff
device \f2type\f1 (see
.BR troff (1)).
Output for a recognized device (\*(mBpost\fP, \*(mBX97.ti10p\fP,
\*(mBX97.ti12p\fP, \*(mBi300\fP, \*(mBpcl\fP)
is sent through the appropriate postprocessor for printing.
The default device type is \*(mBpost\fP.
.TP
.OP \-D dev
directs the output to 
device type \f2dev\f1.
Supported values for \f2dev\f1 are:
.sp 4p
.in +0.75i
.de XX
.sp 2p
.ti -0.75i
\*(mB\\$1\f1
.sp -1v
..
.XX post
PostScript laser printer
.XX lino
Linotronic L200P PostScript-based phototypesetter
.XX i300
Imagen 8/300 laser printer
.XX xrx
Xerox 4050/9700/8700 line printer
.XX hp
Hewlett-Packard LaserJet printer
.in -0.75i
.sp 4p
.TP
.OP \-z
invokes no output filter
to process or redirect the output of
.BR troff .
.PD
.PP
When you specify
.OP \-
instead of any
.IR files ,
.B mmt 
will read the standard input.
.SH EXAMPLES
To process input with
.B troff
and the
.SM MM
macros, and preview the output on a local PostScript printer:
.EX
mmt \-Tpost \-Dpost  \f2files
.EE
Then, to print that document on the phototypesetter:
.EX
mmt \-Tpost \-Dlino  \f2files
.EE
To process input with
.BR tbl ,
.BR eqn ,
.BR pic ,
.BR grap ,
and
.B troff
with the
.SM MM
macros and send output to an Imagen 8/300 printer
via a local spooling command:
.EX
mmt \-g \-p \-t \-e \-Ti300 \-z  \f2files\fP | \f2spool
.EE
Note:  The
.OP \-z
option is needed to allow special postprocessing functions or commands.
.br
.ne 1i
.SH DIAGNOSTICS
.MW "mmt: no input file"
.ti +3n
If none of the arguments is a readable file and
the command
is not used as a filter.
.SH HINTS
Use the
.OP \-o list
option of
.B troff
to specify a subset or range of pages to be output.
Note, however, that 
if you call
.B mmt
with one or more of the
.OP \-e ,
.OP \-t ,
.OP \-p ,
.OP \-g ,
and
.OP \-
options
together
with the
.OP \-o list
option of
.BR troff ,
you may cause a harmless ``broken pipe'' diagnostic
if the last page of the document is not specified in
.IR list .
.SH SEE ALSO
.BR mm (1),
.BR mvt (1),
.BR troff (1),
.BR tbl (1),
.BR eqn (1),
.BR pic (1),
.BR grap (1),
.BR eqnchar (5),
.BR mm (5)


misc/mmt/mmt.sh

PUBDIR=/usr/pub

u=-mm;  if echo `basename $0` | grep mv >/dev/null;  then u=-mv;  fi
if test $# = 0;
then 
cat >&2 <<!
Usage: "mmt [options] files" (or "mvt [options] files") where "options" are:
 -a	=> output to terminal
 -e	=> eqn
 -p	=> pic
 -g	=> grap
 -t	=> tbl
 -Tptty	=> format for troff device ptty (post, i10, i300, pcl, X97.ti10p, X97.ti12p)
 -Dpost => send to PostScript
 -Dlino => send to MHCC Linotronics L200P (PostScript) phototypesetter
 -Di10	=> send to Imagen Imprint-10
 -Di300 => send to Imagen 8/300
 -Dhp   => send to HP LaserJet
 -Dxrx  => send to Xerox 4050,8700,9700
 -z	=> use no postprocessor for troff output
 -	=> instead of "files" inside a pipeline.
Other options as required by TROFF and the macros.
!
exit 1
fi
#	Linotronic L200P (PostScript) phototypesetter:
#		Connected directly, use o="|dpost"
#		To MHCC service, use o="|apsend"	(default)
#	PostScript laser printers:
#		Connected directly, use o="|dpost"
#		Spooled, use locally approved command (e.g., prt)	(default)
#	Imagen (Imprint-10, 8/300) laser printers:
#		Connected directly, use o="|dimpress"	(default, stdout)
#		Spooled, use locally approved command (e.g., i10send, prt)
#	Xerox 4050 Laser Printer
#		Connected directly, use "o=|dx9700 -T$ptty"
#		Spooled, use o="|dx9700 -T$ptty | prt -lraw"	(default)
#	Hewlett-Packard LaserJet laser printers:
#		Connected directly, use o="|dpcl"	(default, stdout)
#		Spooled, use locally approved command (e.g., prt)
#	Other output devices:  Use -Tdevice, -z, and pipe to command		
#
ptty=  x=  Teqn=  ec=${PUBDIR}/eqnchar
PATH=$PATH:/bin:/usr/bin  O=  o=""  y=
a= e= f= p= z= g=
while test -n "$1" -a ! -f "$1"
do case $1 in
	-a)	O="-a"  o=  x=  ;;
	-Tpost)	ptty=post
		o="|prt -ltroff"  x=  ;;
	-Dpost)	o="|prt -ltroff"  x=  ;;
	-Dlino)	o="|apsend"  x=  ;;
	-Ti10)	ptty=i10
		o="|dimpress"  x=  ;;
	-Di10)	o="|dimpress"  x=  ;;
	-Ti300) ptty=i300  ec=${PUBDIR}/i300eqnchar
		o="|dimpress"  ;;
	-Di300)	o="|dimpress"  x=  ;;
	-TX97.ti10p)	ptty=X97.ti10p
			o="|dx9700 -T$ptty |prt -lraw" ;;
	-TX97.ti12p)	ptty=X97.ti12p
			o="|dx9700 -T$ptty |prt -lraw" ;;
	-Dxrx)	o="|dx9700 -TX97.ti10p |prt -lraw"  x=  ;;
	-Tpcl)	ptty=pcl
		o="|dpcl"  x=  ;;
	-Dhp)	o="|dpcl"  x=  ;;
	-T*)	ptty=`echo $1 | sed "s/-T//"`
		o=  x=  ;;
	-e)	e=eqn ;;
	-t)	f=tbl ;;
	-p)	p=pic ;;
	-g)	g=grap  p=pic ;;
	-y)	;;
	-)	break ;;
	-z)	z=z  ;;
	*)	a="$a $1" ;;
   esac
   shift
done
if test -z "$1";  then echo "$0: no input file" >&2;  exit 1;  fi
if test -n "$ptty"
then troff="troff -T$ptty"
else troff=troff
fi
if test -n "$z";  then o=  x=;  fi
if test "|apsend" = "$o";  then x="$x c=$1";  fi
d="$*"
if test -n "$g";  then g="grap $d|";  d="-";  fi
if test -n "$p";  then p="pic $d|";  d="-";  fi
if test -n "$f";  then f="tbl $d|";  d="-";  fi
if test -n "$e";  then e="eqn $ec $d|";  d="-";  fi
if test "$u" != "-mv";  then y=;  fi
eval "$g $p $f $e $troff $y $O $u $a $d $o $x"; exit 0
